pkg/store/ratelimited/ratelimited.go

```diff
@@ -43,19 +43,22 @@ func (s *lstore) writeMetrics(ctx context.Context, p *store.PartitionedMetrics,
 		return nil
 	}
 
+	if limiter := s.limiter(p.PartitionKey); !limiter.AllowN(now, 1) {
```

Comment:
just as a verification point: I rechecked if `AllowN` does any mutations internally, it does, but they are also guarded internally by a mutex: https://github.com/golang/time/blob/master/rate/rate.go#L288.

So this should be fine.